gr-blocks/lib/mute_impl.cc

```diff
@@ -36,41 +37,36 @@ mute_impl<T>::mute_impl(bool mute)
 {
     this->message_port_register_in(pmt::intern("set_mute"));
     this->set_msg_handler(pmt::intern("set_mute"),
-                          [this](pmt::pmt_t msg) { this->set_mute_pmt(msg); });
+                          [this](pmt::pmt_t msg) { this->mute_message_handler(msg); });
 }
 
 template <class T>
 mute_impl<T>::~mute_impl()
 {
 }
 
+template <class T>
+void mute_impl<T>::mute_message_handler(pmt::pmt_t msg)
+{
+    if (pmt::is_pair(msg)) {
+        this->set_mute(pmt::to_bool(pmt::cdr(msg)));
+    } else {
+        this->set_mute(pmt::to_bool(msg));
+    }
+}
+
 template <class T>
 int mute_impl<T>::work(int noutput_items,
                        gr_vector_const_void_star& input_items,
                        gr_vector_void_star& output_items)
 {
-    const T* iptr = (const T*)input_items[0];
-    T* optr = (T*)output_items[0];
-
-    int size = noutput_items;
+    const T* in = (const T*)input_items[0];
+    T* out = (T*)output_items[0];
 
     if (d_mute) {
-        std::fill_n(optr, noutput_items, 0);
+        std::fill_n(out, noutput_items, 0);
     } else {
-        while (size >= 8) {
-            *optr++ = *iptr++;
-            *optr++ = *iptr++;
-            *optr++ = *iptr++;
-            *optr++ = *iptr++;
-            *optr++ = *iptr++;
-            *optr++ = *iptr++;
-            *optr++ = *iptr++;
-            *optr++ = *iptr++;
-            size -= 8;
-        }
-
-        while (size-- > 0)
-            *optr++ = *iptr++;
+        std::copy_n(in, noutput_items, out);
```

Comment:
*much* nicer. Let the compiler vectorize. Manual loop unrolling is probably a very bad idea. :+1: